### DISCUSSION TITLE
Impossible to delete more than 100,000 records even in OLAP mode

### DISCUSSION THREAD

**Discussion by baptisteArno** [planetscale/discussion#388] [Q&A] [answered]
I'm trying to empty a table, but it fails with the message:

```
target: typebot.-.primary: vttablet: rpc error: code = Aborted desc = caller id: oqcm30z7knpwnk7x7mwq: row count exceeded 100000 (CallerID: oqcm30z7knpwnk7x7mwq)
```

It happens even in OLAP workflow mode?

<img width="1647" alt="CleanShot 2023-02-08 at 11 44 12@2x" src="https://user-images.githubusercontent.com/16015833/217508036-90ee4108-c79a-405c-b983-5a6fd871b562.png">

I also tried to do it with a Prisma client with the query param: `sessionVariables=workload=olap` and it fails:

```
PrismaClientUnknownRequestError: 
Invalid `prisma.log.deleteMany()` invocation:


Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Server(ServerError { code: 1105, message: "vttablet: rpc error: code = Aborted desc = Row count exceeded 100000 (CallerID: ji1dvf3js4uje0b63cvg)", state: "HY000" })), transient: false })
    at Object.handleRequestError (/Users/baptistearno/Dev/prisma-database-sync/src/prisma-clients/target/runtime/index.js:31950:13)
    at Object.handleAndLogRequestError (/Users/baptistearno/Dev/prisma-database-sync/src/prisma-clients/target/runtime/index.js:31913:12)
    at Object.request (/Users/baptistearno/Dev/prisma-database-sync/src/prisma-clients/target/runtime/index.js:31908:12)
    at PrismaClient._request (/Users/baptistearno/Dev/prisma-database-sync/src/prisma-clients/target/runtime/index.js:32994:16)
    at <anonymous> (/Users/baptistearno/Dev/prisma-database-sync/src/reset.ts:36:24)
    at async Promise.all (index 16)
    at reset (/Users/baptistearno/Dev/prisma-database-sync/src/reset.ts:30:3)
    at main (/Users/baptistearno/Dev/prisma-database-sync/src/index.ts:4:3) {
  clientVersion: '4.9.0',
  batchRequestIdx: undefined
}
```

I really need to make this work with Prisma because I'm implementing a library that will help user migrate from any database using Prisma. Any idea how to solve this?

**Comment by orware** [comment, marked as answer]
Hello @baptisteArno,

I believe `OLAP` mode may only assist for the `SELECT` side of things in this case, so the solution in this situation would be to ensure you add a `LIMIT` clause to your `DELETE` queries with the 100K limit in mind, and possibly even use a smaller limit too to help ensure that each query doesn't take too long to execute.

You could then issue those `DELETE` queries in a loop until no further rows are affected.

For example:

```sql
-- Run the following query in a loop within Prisma until there are no more affected rows:
DELETE FROM Log LIMIT 10000;
```

If Prisma's `deleteMany()` [method](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#deletemany) doesn't offer the ability to include a `LIMIT` then using an alternative like what `$executeRaw` [provides](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#executeraw) may be required instead.

I hope this solution is helpful!

**Reply by baptisteArno** [reply]
Alright, thank you 👌
